magento/pwa-studio
# Delete Address from Address Book

## Description

This PR adds the ability to delete addresses from the Address Book.

## Related Issue





Closes PWA-635.

## Acceptance



### Verification Stakeholders


@schensley @dpatil-magento 

### Specification



### Verification Steps



1. Sign in
1. Go to the Address Book page
1. Verify that a "Delete" button appears for addresses that are not the default address
1. Verify that clicking the "Delete" button presents a confirmation overlay on the target address
1. Click "Cancel"
1. Verify that the confirmation overlay disappears and nothing happens
1. Click the "Delete" button again
1. Click "Delete" in the confirmation overlay
1. Verify that the buttons are disabled for a second while the request is in flight
1. Verify that the target address is deleted and the page refreshes<sup>1</sup>

<sup>1</sup>: There is a slight delay after the overlay disappears and the address actually gets removed. This is due to a workaround due to a graphQL bug and will be resolved in another scope.

## Screenshots / Screen Captures (if appropriate)

![Screen Shot 2020-12-03 at 12 15 53 PM](https://user-images.githubusercontent.com/13182778/101070759-648d1700-3561-11eb-8768-0379daac34be.png)

![Screen Shot 2020-12-03 at 12 16 01 PM](https://user-images.githubusercontent.com/13182778/101070763-65be4400-3561-11eb-968f-20d85f18c9b7.png)

Mobile:
![Screen Shot 2020-12-03 at 12 16 21 PM](https://user-images.githubusercontent.com/13182778/101070761-6525ad80-3561-11eb-93a6-65dd23740175.png)


## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.

- [x] The confirmation buttons extend past their container sometimes
